### PR TITLE
fix #41934

### LINKED ISSUE #41934: [Package Request]: Hashicorp Terraform
### Package Requested

- [ ] I would like someone else to build the manifest.
- [ ] I would like help so I can submit the manifest.
- [X] I have performed a search and couldn't find this package.
- [ ] I have checked that the installer for this package is not a standalone or portable executable or a .zip file
- [ ] I think there is a new version available and I have provided the URL.

### Please provide the following information

Publisher: Hashicorp
Publisher Url: https://www.terraform.io/
Package Name: Hashicorp.Terraform
Description: Terraform is an open-source infrastructure as code software tool that provides a consistent CLI workflow to manage hundreds of cloud services. Terraform codifies cloud APIs into declarative configuration files.
Package Version: 1.1.4
Installer URL: https://releases.hashicorp.com/terraform/1.1.4/terraform_1.1.4_windows_amd64.zip

Current Package Identifier (for new versions): 

## Note:
There is currently no MSI installer, only a zip with a single-exe.

### PR DESCRIPTION
**Pull request by DanielHabenicht**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [ ] Does your manifest conform to the [1.2 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.2.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/94218)

### CODE REVIEW

**Comment by stephengillie** [comment]
When manually validating these, it's an odd situation. Downloaded and extracted, these run as expected. But when installed through Winget, they generate errors:
```
PS C:\WinGet\Scripts\Executables> C:\Users\WDAGUtilityAccount\AppData\Local\Microsoft\WinGet\Links\terraform.exe
Program 'terraform.exe' failed to run: The specified executable is not a valid application for this OS platform.

PS C:\WinGet\Scripts\Executables> C:\Users\WDAGUtilityAccount\AppData\Local\Microsoft\WinGet\Packages\Hashicorp.Terraform__DefaultSource\terraform_1.3.7_windows_amd64.exe
Program 'terraform_1.3.7_windows_amd64.exe' failed to run: The specified executable is not a valid application for
this OS platform.
```

**Comment by DanielHabenicht** [comment]
Yes, I had the same problem, but did not had the time to investigate. 

It worked over in my 1Password CLI PR (which also uses zip)

**Comment by DanielHabenicht** [comment]
If this goes on like this I think I will update the PR at #95950 with my changes

**Comment by kdpuvvadi** [comment]
> If this goes on like this I think I will update the PR at #95950 with my changes

`InstallerType` type should be `zip`

File: manifests/h/Hashicorp/Terraform/1.3.7/Hashicorp.Terraform.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Created with YamlCreate.ps1 v2.2.1 $debug=NVS0.CRLF.5-1-22621-963.Win32NT
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.2.0.schema.json
+
+PackageIdentifier: Hashicorp.Terraform
+PackageVersion: 1.3.7
+InstallerLocale: en-US
+InstallerType: portable
```
Comment (kdpuvvadi) [inline manifests/h/Hashicorp/Terraform/1.3.7/Hashicorp.Terraform.installer.yaml:7]: `InstallerType` should be zip

**Review by kdpuvvadi — CHANGES_REQUESTED** [review]

File: manifests/h/Hashicorp/Terraform/1.3.7/Hashicorp.Terraform.installer.yaml (hunk 1 above)
Comment (kdpuvvadi) [reply manifests/h/Hashicorp/Terraform/1.3.7/Hashicorp.Terraform.installer.yaml:7]: `InstallerType: portable`

**Review by stephengillie — APPROVED** [review]

**Pull request merged**